### ISSUE TITLE
Always full 100% sram measurement

### ISSUE THREAD

**Issue by Testato** [abcminiuser/stackchecker#2]
Hi Dean,

I used your stack checker on a standard C led blink application and all work (32u4 mcu on AtmelStudio 6.2)

Instead if I use the VisualMicro plugin for debug directly an Arduino sketch like the arduino blink code, the checker return always 100% occupied.
Maybe your plugin can be adapted to work whit VisualMicro also ?

Bie

**Comment by abcminiuser** [comment]
Hi testato,

Can you send me a memory dump of a program running on your board, that was built using the VisualMicro plugin? For my extension to work you must use a debugger (not a programmer) and the plugin must be able to dump the entire device RAM.

Can you confirm that the stack instrumentation code is being run on startup with your application? I suspect that the C++ initialization runtime does not give the same early init hooks that the regular C avr-libc runtime does which my code is reliant upon.

**Comment by Testato** [comment]
yes,  i know that your plugin work in debugging mode (i writed that it work normally on C code)

Instead whit Visualmicro sketch the plugin start when i start debug, after I pause the debug and press the refresh button on your plugin and the bar go to totally green, and it write 100% occupied.

For memory dump it is good an .elf created whit AtmelStudio ?

**Comment by Testato** [comment]
I shared this for a 328P mcu, whit visualmicro and your plugin installed
https://drive.google.com/file/d/0Bw27MYXTb2FXeUQ2NWFsSUZKSmM/edit?usp=sharing

**Comment by abcminiuser** [comment]
Just tried to install the extension; after it threw up a fake Windows security dialog on first boot (always a good sign of a quality product) I'm blocked by the trial button returning a server error so I can't start a new trial period.

I just had a quick chat with my old Atmel Studio developer buddy and he tells me the plugin's debugger doesn't use the normal Atmel Studio debugger internals, so my extension won't be able to work with it. Sorry testato!

**Comment by Testato** [comment]
Hi Dean,
Wait, there is a misunderstanding. VisualMicro have two way of work, the
first is free end only make possible use Arduino sketch .ino on
AtmelStudio. The second is under payment and use a personal way of debug
(simple printing method on serial) that i do not use.
So the free use only adapt the path for the arduino core file, change
simbol name, etcetera.

In this condition, the free condition, and the use of a real debugger, i
use Atmel ICE via jtag on a 32u4, there is no difference from a simple C
blink program and a blink .ino sketch.

The file that i shared for you is an .elf that you can open directly in
Object mode, or you can create one via VisualMicro without activate the
serialprint debugger trial. Simply install VisualMicro, press NO to the
Trial debugger mode, and open an arduino blink.ino sketch.

Thanks for your support, the StackCecker it is a real useful plugin and i
think all arduino users will love it if it will work.

**Comment by Testato** [comment]
news on this bug ?
May you re-open this issue ?

**Comment by abcminiuser** [comment]
Sorry Testato, I haven't had time to look into this recently, but I'll re-open it until a solution is found.
